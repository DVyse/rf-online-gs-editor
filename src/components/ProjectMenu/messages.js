/*
 * ProjectMenu Messages
 *
 * This contains all the text for the ProjectMenu component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.ProjectMenu.header',
    defaultMessage: 'This is the ProjectMenu component !',
  },
  General: {
    id: 'app.components.ProjectMenu.General',
    defaultMessage: 'General',
  },
  Dashboard: {
    id: 'app.components.ProjectMenu.Dashboard',
    defaultMessage: 'Dashboard',
  },
  Contributors: {
    id: 'app.components.ProjectMenu.Contributors',
    defaultMessage: 'Contributors',
  },
  Administration: {
    id: 'app.components.ProjectMenu.Administration',
    defaultMessage: 'Administration',
  },
  Manage: {
    id: 'app.components.ProjectMenu.Manage',
    defaultMessage: 'Manage',
  },
  Items: {
    id: 'app.components.ProjectMenu.Items',
    defaultMessage: 'Items',
  },
  Stores: {
    id: 'app.components.ProjectMenu.Stores',
    defaultMessage: 'Stores',
  },
  UpdateDetails: {
    id: 'app.components.ProjectMenu.UpdateDetails',
    defaultMessage: 'Update Details',
  },
  RemoveProject: {
    id: 'app.components.ProjectMenu.RemoveProject',
    defaultMessage: 'Remove Project',
  },
  ImportExportFiles: {
    id: 'app.components.ProjectMenu.ImportExportFiles',
    defaultMessage: 'Import / Export',
  },
  ImportFiles: {
    id: 'app.components.ProjectMenu.ImportFiles',
    defaultMessage: 'Import Files',
  },
  ExportFiles: {
    id: 'app.components.ProjectMenu.ExportFiles',
    defaultMessage: 'Export Files',
  },
});
