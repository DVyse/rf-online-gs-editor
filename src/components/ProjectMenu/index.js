/**
 *
 * ProjectMenu
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function ProjectMenu({ projectId, project, currentUser }) {
  const isCurrentIsOwner =
    project &&
    currentUser &&
    currentUser.get('id') === project.getIn(['owner', 'id']);

  return (
    <Aside>
      <p className="menu-label">
        <FormattedMessage {...messages.General} />
      </p>
      <ul className="menu-list">
        <li>
          <Link exact to={`/project/${projectId}`}>
            <FormattedMessage {...messages.Dashboard} />
          </Link>
        </li>
        <li>
          <Link to={`/project/${projectId}/contributors`}>
            <FormattedMessage {...messages.Contributors} />
          </Link>
        </li>
      </ul>
      {isCurrentIsOwner && (
        <React.Fragment>
          <p className="menu-label">
            <FormattedMessage {...messages.Manage} />
          </p>
          <ul className="menu-list">
            <li>
              <Link to={`/project/${projectId}/items`}>
                <FormattedMessage {...messages.Items} />
                &nbsp;
                <span className="tag is-small is-primary">
                  {project.getIn(['items', 'total'])}
                </span>
              </Link>
            </li>
          </ul>
          <p className="menu-label">
            <FormattedMessage {...messages.Administration} />
          </p>
          <ul className="menu-list">
            <li>
              <Link to={`/project/${projectId}/details`}>
                <FormattedMessage {...messages.UpdateDetails} />
              </Link>
            </li>
            <li>
              <Link to={`/project/${projectId}/remove`}>
                <FormattedMessage {...messages.RemoveProject} />
              </Link>
            </li>
          </ul>
          <p className="menu-label">
            <FormattedMessage {...messages.ImportExportFiles} />
          </p>
          <ul className="menu-list">
            <li>
              <Link to={`/project/${projectId}/import`}>
                <FormattedMessage {...messages.ImportFiles} />
              </Link>
            </li>
            <li>
              <Link to={`/project/${projectId}/export`}>
                <FormattedMessage {...messages.ExportFiles} />
              </Link>
            </li>
          </ul>
        </React.Fragment>
      )}
    </Aside>
  );
}

ProjectMenu.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  projectId: PropTypes.string.isRequired,
  project: PropTypes.instanceOf(Map),
  currentUser: PropTypes.instanceOf(Map),
};

ProjectMenu.defaultProps = {
  project: null,
  currentUser: null,
};

export default ProjectMenu;

const Link = styled(NavLink).attrs({ activeClassName: 'is-active' })``;
const Aside = styled.aside.attrs({ className: 'menu' })`
  background: #fff;
  padding: 15px;
  border-radius: 3px;
  height: calc(100vh - 73.5px);
  overflow-y: auto;
  box-shadow: 0 -3px 31px 0 rgba(0, 0, 0, 0.05),
    0 6px 20px 0 rgba(0, 0, 0, 0.02);
`;
