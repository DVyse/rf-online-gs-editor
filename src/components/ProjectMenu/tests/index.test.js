import React from 'react';
import { shallow } from 'enzyme';

import ProjectMenu from '../index';

const projectImportsProcessingData = {
  isProcessing: false,
  countTotal: 0,
  countCompleted: 0,
  countProcesses: 0,
  percent: 0,
};

describe('<ProjectMenu />', () => {
  it('Should exist', () => {
    const wrapper = shallow(
      <ProjectMenu
        projectId="123"
        projectImportsProcessingData={projectImportsProcessingData}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
