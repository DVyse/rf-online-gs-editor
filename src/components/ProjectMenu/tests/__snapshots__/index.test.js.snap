// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectMenu /> Should exist 1`] = `
<ProjectMenu__Aside
  vertical={true}
>
  <MenuItem
    header={true}
  >
    <FormattedMessage
      defaultMessage="General"
      id="app.components.ProjectMenu.General"
      values={Object {}}
    />
  </MenuItem>
  <MenuItem
    as={[Function]}
    exact={true}
    to="/project/123"
  >
    <FormattedMessage
      defaultMessage="Dashboard"
      id="app.components.ProjectMenu.Dashboard"
      values={Object {}}
    />
  </MenuItem>
  <MenuItem
    as={[Function]}
    exact={true}
    to="/project/123/contributors"
  >
    <FormattedMessage
      defaultMessage="Contributors"
      id="app.components.ProjectMenu.Contributors"
      values={Object {}}
    />
  </MenuItem>
</ProjectMenu__Aside>
`;
